binding-rust/src/lib.rs

```diff
@@ -53,7 +56,7 @@ impl Space {
 				vec![(rng.gen::<u64>()%n) as f64]
 			},
 			Space::BOX{ref shape, ref high, ref low} => {
-				let mut ret = Vec::new();
+				let mut ret = Vec::with_capacity(shape.iter().map(|x| *x as usize).product());
```

Comment:
IIUC, this doesn't handle multi-dimension arrays correctly. `product(shape)` is the correct size for the vector, but the loop below will only generate `sum(shape)` numbers.